### ISSUE TITLE
CurrentUser modifies http.Request

### ISSUE THREAD

**Issue by orian** [go-authboss/authboss#70]
Authboss.CurrentUser calls Authboss.ContextFromRequest which calls Request.ParseForm. The authentication should be possible with no request modification or it should explicitly warn about parsing in a documentation.

An option would be to allow get a user through another function so it would not break API.

I have a scenario:
A PUT request /entity/12 which have to be authenticated. The request has a specific prefix which allows to handle it efficiently by one of a different data handlers. The body may be multiple megabytes and it may not be a proper value for form.

**Comment by aarondl** [comment]
An oversight to be sure. Will evaluate to see how we can fix this. Expect something around the weekend, busy until then.

**Comment by aarondl** [comment]
Hi @orian. As promised I've worked on this issue over this weekend. I've got a branch for you to try out. Because it's a "bigger" change I'd like to hand it to you first to see if it works for your use case, if not let's try to improve it. From what I can see `CurrentUser` no longer touches the `http.Request.Body`.

https://github.com/go-authboss/authboss/tree/context-request-separation

Once you're satisfied that this branch is the solution, I'll document the changes and we'll get it merged into master. One reason I'm asking you to try this is isolation is to find any unpleasant bugs the refactoring may have created. Though the tests all pass, and the sample appears unaffected it's still a concern as authboss is still awaiting version 1.0. Sorry for that! :)

Also thanks very much for reporting this, and using Authboss!
